examples/rmrk/resource/wasm/tests/resources.rs

```diff
@@ -115,6 +118,31 @@ async fn adding_resource_to_storage_by_admin_succeeds_async() {
     assert_eq!(expected_reply, reply);
 }
 
+#[tokio::test]
+async fn adding_resource_to_storage_by_admin_via_client_succeeds() {
+    // Arrange
+    let fixture = Fixture::new(ADMIN_ID);
+    let catalog_program_id = ActorId::from(fixture.catalog_program().id().into_bytes());
+    let resource_program_id = ActorId::from(fixture.resource_program_for_async().id().into_bytes());
+
+    // Act
+    let resource = resource_client::Resource::Composed(resource_client::ComposedResource {
+        src: "<src_uri>".into(),
+        thumb: "<thumb_uri>".into(),
+        metadata_uri: "<metadata_uri>".into(),
+        base: catalog_program_id,
+        parts: vec![1, 2, 3],
+    });
+    let add_reply = fixture
+        .add_resource_via_client(ADMIN_ID, RESOURCE_ID, resource, resource_program_id)
```

Comment:
nit: resource_program_id can be taken and passed inside the method as you retrieve it from the fixture anyway